### PR TITLE
InputEndpointImpl method implementation and javadoc.

### PR DESCRIPTION
**Pull request by anu3990**
So we can incorporate your pull request, please share the following:
* What issue are you addressing with this pull request? https://project.marklogic.com/jira/browse/JAVA-85 and https://project.marklogic.com/jira/browse/JAVA-87
* Are you modifying the correct branch? (See CONTRIBUTING.md)
* Have you run unit tests? (See CONTRIBUTING.md)
* Version of MarkLogic Java Client API (see Readme.txt)
* Version of MarkLogic Server (see admin gui on port 8001)
* Java version (`java -version`)
* OS and version
* What Changed: What happened before this change? What happens without this change?

### CODE REVIEW

File: marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java [hunk 1]
```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
 
 		@Override
 		public void accept(InputStream input) {
-			// TODO Auto-generated method stub
+			
+			InputStream output = null;
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java]: Would it be possible to declare the variable at the point of assignment on line 130?

If so, that might make the code easier to understand for someone else who maintains it.

File: marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java [hunk 2]
```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
...
+			try {
+				queue.put(input);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java]: If the bulk caller can't accept the input, is that a condition that the caller would need to know about?

Would it make sense to rethrow the exception as a wrapped runtime exception?

File: marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java [hunk 3]
```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
...
+					break;
+				case RUNNING:
+					break;
+				case INTERRUPTING:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case INTERRUPTED:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case COMPLETED:
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java]: Do these cases need separate throw actions, given that cases fall through in Java?

Also, is it accurate to report the problem as an internal exception?  Would a Java IllegalStateException be more accurate?

File: marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java [hunk 4]
```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
...
+			
+			logger.trace("input endpoint running endpoint={} count={} state={}",
+                    getEndpointPath(), getCallCount(), getEndpointState());
+			output = getEndpoint().getCaller().call(
+					getEndpoint().getClient(), getEndpointState(), session, getWorkUnit(), inputStreamList.stream()
+					);
+			
+			if (allowsEndpointState()) {
+                setEndpointState(output);
+            }
```
Comment (ehennum) [inline marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java]: If the indentation were more regular, the code might be easier for someone else to maintain.

**Comment by anu3990** [comment]
Added the suggestions and adjusted the indentation.

**Pull request merged**